File: test1/src/main/java/com/example/test1/API/RestApi.java
package com.example.test1.API;

import android.content.Context;
import android.database.Observable;

import com.google.gson.JsonObject;

import java.util.concurrent.TimeUnit;

import okhttp3.OkHttpClient;
import retrofit2.Retrofit;
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
import retrofit2.converter.gson.GsonConverterFactory;

/**
 * Created by Niken on 7/22/2020.
 */
public class RestApi {
    Context context;
    ApiServices apiServices;
    private static OkHttpClient client;
    private static Retrofit retrofit;

    public RestApi(Context context) {
        this.context = context;
    }

    public void init(){
         retrofit = new Retrofit.Builder()
                .baseUrl(ApiUrl.Initiate)
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .addConverterFactory(GsonConverterFactory.create())
                .client(client(context))
                .build();

         apiServices = retrofit.create(ApiServices.class);
    }

    private OkHttpClient client(Context context) {
        if(client == null){
            client = new OkHttpClient.Builder()
                    .readTimeout(60, TimeUnit.SECONDS)
                    .writeTimeout(60, TimeUnit.SECONDS)
                    .build();
        }
        return client;
    }

}

File: app/src/main/java/com/example/intentapp/API/RestApi.kt
package com.example.intentapp.API

import okhttp3.Interceptor
import okhttp3.OkHttpClient
import okhttp3.Request
import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory
import java.util.concurrent.TimeUnit

/**
 * Created by Niken on 7/23/2020.
 */
object RestApi {
    val apiServices: ApiServices

    init {

        val builder = OkHttpClient.Builder()
            .readTimeout(60, TimeUnit.SECONDS)
            .connectTimeout(60, TimeUnit.SECONDS)
            .writeTimeout(180, TimeUnit.SECONDS)

//        if (PreferenceHelper.containsAccessToken()) {
//            builder.addInterceptor(Interceptor() {
//                val original: Request = it.request()
//                val request: Request = original.newBuilder()
//                    .header("Authorization", "Bearer ${PreferenceHelper.getAccessToken()}")
//                    .method(original.method, original.body)
//                    .build()
//
//                it.proceed(request)
//            })
//
//        }

        val okHttpCLient = builder.build()

        val retrofit = Retrofit.Builder()
            .baseUrl(ApiUrl.BaseUrl)
            .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
            .addConverterFactory(GsonConverterFactory.create())
            .client(okHttpCLient)
            .build()

        apiServices = retrofit.create(ApiServices::class.java)
    }

    fun getApiService(): ApiServices = apiServices
}